### PR TITLE
forex: float support

### PR DESCRIPTION
**Pull request by SoggySaussages**
Adds support for float input to forex

Before:
Screenshot 2022-11-23 at 10 00 07

After:
Screenshot 2022-11-23 at 10 00 15

### CODE REVIEW

**Comment by SoggySaussages** [comment]
Noticing now that float conversion error should return error as message instead of throwing generic command error

**Comment by SoggySaussages** [comment]
Added custom response :)

**Comment by jo3-l** [comment]
Is there a reason you aren't using `dcmd.Float`?

File: lib/dcmd/arg.go [hunk 1]
```diff
@@ -128,6 +128,36 @@ func (p *ParsedArg) Int64() int64 {
...
+		return float64(t)
+	case uint:
+		return float64(t)
+	case uint32:
+		return float64(t)
+	case uint64:
+		return float64(t)
+	case string:
+		parsed, _ := strconv.ParseFloat(t, 64)
+		return parsed
```
Comment (jo3-l) [inline lib/dcmd/arg.go:155]: Remove this case for consistency with `ParsedArg.Int`: it's not `Float64`'s responsibility to parse string values, As is, this also discards any parse errors, which is inadvisable.
```suggestion
```

File: stdcommands/forex/forex.go [hunk 2]
```diff
@@ -41,7 +42,7 @@ var Command = &commands.YAGCommand{
 		if (to == nil) || (from == nil) {
 			return "Invalid currency code.\nCheck out available codes on: <https://api.exchangerate.host/symbols>", nil
 		}
-		output, err := requestAPI("https://api.exchangerate.host/convert?from=" + from.Code + "&to=" + to.Code + "&amount=" + amount.Str())
+		output, err := requestAPI("https://api.exchangerate.host/convert?from=" + from.Code + "&to=" + to.Code + "&amount=" + strconv.FormatFloat(amount, 'f', -1, 64))
```
Comment (jo3-l) [inline stdcommands/forex/forex.go:45]: This may be better written as a call to Sprintf.

**Review by jo3-l — CHANGES_REQUESTED** [review]

File: stdcommands/forex/forex.go (hunk 2 above)
Comment (SoggySaussages) [reply stdcommands/forex/forex.go]: Sounds good, thanks

**Pull request merged**